### PR TITLE
Update F# on Azure Docs - Queue Storage

### PR DESCRIPTION
**Pull request by zedy-wj**
1.Updating F# doc to the new SDKs - Queue Storage

2.Updated files : 

- i.docs/fsharp/using-fsharp-on-azure/queue-storage.md
- ii.samples/snippets/fsharp/azure/queue-storage.fsx

@jongio for notification.

### CODE REVIEW

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md [hunk 1]
```diff
@@ -30,67 +30,53 @@ Next, use a [package manager](package-management.md) such as [Paket](https://fsp
...
+[!code-fsharp[QueueStorage](../../../samples/snippets/fsharp/azure/queue-storage.fsx#L19-L20)]
 
 Now you are ready to write code that reads data from and writes data to Queue storage.
 
-## Create a queue
-
-This example shows how to create a queue if it doesn't already exist:
-
-[!code-fsharp[QueueStorage](~/samples/snippets/fsharp/azure/queue-storage.fsx#L32-L36)]
-
```
Comment (jongio) [inline docs/fsharp/using-fsharp-on-azure/queue-storage.md:72]: Why was this removed?

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md [hunk 2]
```diff
@@ -124,40 +110,40 @@ software failure, another instance of your code can get the same message
...
+[!code-fsharp[QueueStorage](../../../samples/snippets/fsharp/azure/queue-storage.fsx#L68-L77)]
 
 ## Additional options for de-queuing messages
 
 There are two ways you can customize message retrieval from a queue.
 First, you can get a batch of messages (up to 32). Second, you can set a
 longer or shorter invisibility timeout, allowing your code more or less
 time to fully process each message. The following code example uses
-`GetMessages` to get 20 messages in one call and then processes
+`ReceiveMessages` to get 20 messages in one call and then processes
```
Comment (jongio) [inline docs/fsharp/using-fsharp-on-azure/queue-storage.md:127]: Is this still true that it will get 20 messages?

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md [hunk 3]
```diff
@@ -124,40 +110,40 @@ software failure, another instance of your code can get the same message
...
 each message. It also sets the invisibility timeout to five minutes for
 each message. The 5 minutes starts for all messages at the same
-time, so after 5 minutes have passed since the call to `GetMessages`, any
+time, so after 5 minutes have passed since the call to `ReceiveMessages`, any
```
Comment (jongio) [inline docs/fsharp/using-fsharp-on-azure/queue-storage.md:130]: Is the default timeout still 5 min?

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md [hunk 4]
```diff
@@ -30,67 +30,53 @@ Next, use a [package manager](package-management.md) such as [Paket](https://fsp
 
 Add the following `open` statements to the top of the `queues.fsx` file:
 
-[!code-fsharp[QueueStorage](~/samples/snippets/fsharp/azure/queue-storage.fsx#L1-L3)]
+[!code-fsharp[QueueStorage](../../../samples/snippets/fsharp/azure/queue-storage.fsx#L1-L2)]
```
Comment (jongio) [inline docs/fsharp/using-fsharp-on-azure/queue-storage.md:33]: Why were all the paths changed from ~/ to ../../../?

**Review by jongio — CHANGES_REQUESTED** [review]

File: samples/snippets/fsharp/azure/queue-storage.fsx [hunk 5]
```diff
@@ -12,102 +11,88 @@ let storageConnString = "..." // fill this in from your storage account
...
+queueClient.CreateIfNotExists()
 
 //
 // Insert a message into a queue.
 //
 
 // Create a message and add it to the queue.
-let message = new CloudQueueMessage("Hello, World")
-queue.AddMessage(message)
+let message = "Hello, World"
```
Comment (jsquire) [inline samples/snippets/fsharp/azure/queue-storage.fsx:27]: Given that this is a primitive now, it may be better to collapse 27 and 28 into a single line.    The comment on L26 should also be adjusted.

Is this restricted to `string` or is there a more complex flow that we should also consider demonstrating?

File: samples/snippets/fsharp/azure/queue-storage.fsx [hunk 6]
```diff
@@ -12,102 +11,88 @@ let storageConnString = "..." // fill this in from your storage account
...
 
 //
 // Peek at the next message.
 //
 
 // Peek at the next message.
-let peekedMessage = queue.PeekMessage()
-let msgAsString = peekedMessage.AsString
+let peekedMessage = queueClient.PeekMessage()
+let msgToString = peekedMessage.ToString()
```
Comment (jsquire) [inline samples/snippets/fsharp/azure/queue-storage.fsx:36]:
```suggestion
let messageContents = peekedMessage.ToString()
```

File: samples/snippets/fsharp/azure/queue-storage.fsx [hunk 7]
```diff
@@ -12,102 +11,88 @@ let storageConnString = "..." // fill this in from your storage account
...
 //
 // Use Async-Await pattern with common Queue storage APIs.
 //
 
 async {
-    let! exists = queue.CreateIfNotExistsAsync() |> Async.AwaitTask
+    let! exists = queueClient.CreateIfNotExistsAsync() |> Async.AwaitTask
 
-    let! retrieved = queue.GetMessageAsync() |> Async.AwaitTask
+    let! receipt = queueClient.ReceiveMessageAsync() |> Async.AwaitTask
```
Comment (jsquire) [inline samples/snippets/fsharp/azure/queue-storage.fsx:70]: I'm not sure why we're using different nomenclature for a received message:

- `receipt` here
- `retrieved` on L43
- `peekedMessage` on L35
- `message` for a string on L27
- `msg` on L82

We should consider normalizing terminology unless we've got this many types to represent the same concept.

File: samples/snippets/fsharp/azure/queue-storage.fsx [hunk 8]
```diff
@@ -12,102 +11,88 @@ let storageConnString = "..." // fill this in from your storage account
...
-queue.FetchAttributes()
-let count = queue.ApproximateMessageCount.GetValueOrDefault()
+let properties = queueClient.GetProperties().Value
+let count = properties.ApproximateMessagesCount
 
 //
 // Delete a queue.
 //
 
 // Delete the queue.
```
Comment (jsquire) [inline samples/snippets/fsharp/azure/queue-storage.fsx:97]:
```suggestion
```

This comment doesn't add any context to the block on L93-95,  may as well delete it.

File: samples/snippets/fsharp/azure/queue-storage.fsx [hunk 9]
```diff
@@ -12,102 +11,88 @@ let storageConnString = "..." // fill this in from your storage account
...
 //
 // Change the contents of a retrieved message.
 //
 
 // Update the message contents and set a new timeout.
-retrieved.SetMessageContent("Updated contents.")
-queue.UpdateMessage(retrieved,
-    TimeSpan.FromSeconds(60.0),
-    MessageUpdateFields.Content ||| MessageUpdateFields.Visibility)
+queueClient.UpdateMessage(
```
Comment (jsquire) [inline samples/snippets/fsharp/azure/queue-storage.fsx:50]: If the call on L43 fails, what happens when someone tries to reference the `Value` property?   If seems like we should discuss and cover it as a failure scenario.

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md (hunk 2 above)
Comment (jsquire) [reply docs/fsharp/using-fsharp-on-azure/queue-storage.md:127]: Looks that way from the code snippet.

File: samples/snippets/fsharp/azure/queue-storage.fsx [hunk 10]
```diff
@@ -12,102 +11,88 @@ let storageConnString = "..." // fill this in from your storage account
...
-//
-// Create a queue.
-//
-
-// Retrieve a reference to a container.
-let queue = queueClient.GetQueueReference("myqueue")
-
-// Create the queue if it doesn't already exist
-queue.CreateIfNotExists()
+let queueClient = new QueueClient(storageConnString, "myqueue");
```
Comment (cartermp) [inline samples/snippets/fsharp/azure/queue-storage.fsx:19]:
```suggestion
let queueClient = QueueClient(storageConnString, "myqueue");
```

Unless this is disposable, you can omit the `new`

Comment (zedy-wj) [reply samples/snippets/fsharp/azure/queue-storage.fsx:19]: Thanks for your guidance!

File: samples/snippets/fsharp/azure/queue-storage.fsx (hunk 5 above)
Comment (zedy-wj) [reply samples/snippets/fsharp/azure/queue-storage.fsx:27]: Thanks for your guidance! I have adjust it.

Parameter `message` in the method `AddMessage` is only support two types `string` and `BinaryData` now.

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md (hunk 4 above)
Comment (zedy-wj) [reply docs/fsharp/using-fsharp-on-azure/queue-storage.md:33]: Because all the paths ~/ could not link to code page.

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md (hunk 1 above)
Comment (zedy-wj) [reply docs/fsharp/using-fsharp-on-azure/queue-storage.md]: Put it together with `Create the Queue Service client.`. It has been token back now.

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md (hunk 3 above)
Comment (zedy-wj) [reply docs/fsharp/using-fsharp-on-azure/queue-storage.md:136]: Yes, it was set [here](https://github.com/zedy-wj/docs/blob/queue-storage/samples/snippets/fsharp/azure/queue-storage.fsx#L88).

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md (hunk 2 above)
Comment (zedy-wj) [reply docs/fsharp/using-fsharp-on-azure/queue-storage.md:133]: Yes, it was set [here](https://github.com/zedy-wj/docs/blob/queue-storage/samples/snippets/fsharp/azure/queue-storage.fsx#L88).

File: samples/snippets/fsharp/azure/queue-storage.fsx [hunk 11]
```diff
@@ -12,102 +11,93 @@ let storageConnString = "..." // fill this in from your storage account
...
-
-// Parse the connection string and return a reference to the storage account.
-let storageAccount = CloudStorageAccount.Parse(storageConnString)
 
 //
 // Create the Queue Service client.
 //
 
-let queueClient = storageAccount.CreateCloudQueueClient()
+// Instantiate a QueueClient which will be used to create and manipulate the queue
```
Comment (jsquire) [inline samples/snippets/fsharp/azure/queue-storage.fsx:19]: It seems that the first comment after the 3-line header block for each section just repeats what the header mentions.   We should probably consider collapsing them; they're pretty redundant.

File: samples/snippets/fsharp/azure/queue-storage.fsx [hunk 12]
```diff
@@ -12,102 +11,93 @@ let storageConnString = "..." // fill this in from your storage account
...
+    "Updated contents.",
+    TimeSpan.FromSeconds(60.0))
 
 //
 // De-queue the next message, indicating successful processing
 //
 
 // Process the message in less than 30 seconds, and then delete the message.
-queue.DeleteMessage(retrieved)
+let delMessage = queueClient.ReceiveMessage().Value
```
Comment (jsquire) [inline samples/snippets/fsharp/azure/queue-storage.fsx:66]:
```suggestion
let deleteMessage = queueClient.ReceiveMessage().Value
```

**Review by jsquire — APPROVED** [review]
LGTM

**Comment by zedy-wj** [comment]
Complete all the changes according to comments. Thanks for your guidance!

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md [hunk 13]
```diff
@@ -30,67 +30,59 @@ Next, use a [package manager](package-management.md) such as [Paket](https://fsp
...
-To parse the connection string, use:
-
-[!code-fsharp[QueueStorage](~/samples/snippets/fsharp/azure/queue-storage.fsx#L19-L20)]
-
-This will return a `CloudStorageAccount`.
-
 ### Create the Queue service client
 
-The `CloudQueueClient` class enables you to retrieve queues stored in Queue storage. Here's one way to create the service client:
+The `QueueClient` class enables you to retrieve queues stored in Queue storage. Here's one way to create the service client:
```
Comment (tg-msft) [inline docs/fsharp/using-fsharp-on-azure/queue-storage.md:53]: There's a separate type called `QueueServiceClient` so let's drop the word "service".

```suggestion
The `QueueClient` class enables you to retrieve queues stored in Queue storage. Here's one way to create the client:
```

File: samples/snippets/fsharp/azure/queue-storage.fsx [hunk 14]
```diff
@@ -12,102 +11,87 @@ let storageConnString = "..." // fill this in from your storage account
...
 
 //
 // Peek at the next message.
 //
 
-// Peek at the next message.
-let peekedMessage = queue.PeekMessage()
-let msgAsString = peekedMessage.AsString
+let peekedMessage = queueClient.PeekMessage()
+let messageContents = peekedMessage.ToString()
```
Comment (tg-msft) [inline samples/snippets/fsharp/azure/queue-storage.fsx:38]: Have you tested this?  We don't override `ToString` so I don't think you'll get anything good out here.  At a minimum, I think you want `peekedMessage.Value.Body.ToString()`.

File: samples/snippets/fsharp/azure/queue-storage.fsx [hunk 15]
```diff
@@ -12,102 +11,87 @@ let storageConnString = "..." // fill this in from your storage account
...
+let properties = queueClient.GetProperties().Value
+let count = properties.ApproximateMessagesCount
 
 //
 // Delete a queue.
 //
 
-// Delete the queue.
-queue.Delete()
+queueClient.Delete()
```
Comment (tg-msft) [inline samples/snippets/fsharp/azure/queue-storage.fsx:97]: Prefer `DeleteIfExists` over `Delete`.

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md [hunk 16]
```diff
@@ -30,67 +30,59 @@ Next, use a [package manager](package-management.md) such as [Paket](https://fsp
...
 
 ## Insert a message into a queue
 
 To insert a message into an existing queue, first create a new
-`CloudQueueMessage`. Next, call the `AddMessage` method. A
-`CloudQueueMessage` can be created from either a string (in UTF-8
-format) or a `byte` array, like this:
+Message. Next, call the `SendMessage` method. A
+Message can be created from string (in UTF-8
+format), like this:
```
Comment (tg-msft) [inline docs/fsharp/using-fsharp-on-azure/queue-storage.md:70]: It might be worth showing off different examples with [BinaryData](https://docs.microsoft.com/en-us/dotnet/api/overview/azure/system.memory.data-readme) because JSON is the most common customer use case for messages.

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md [hunk 17]
```diff
@@ -124,40 +116,40 @@ software failure, another instance of your code can get the same message
 and try again. Your code calls `DeleteMessage` right after the message
 has been processed.
 
-[!code-fsharp[QueueStorage](~/samples/snippets/fsharp/azure/queue-storage.fsx#L75-L76)]
+[!code-fsharp[QueueStorage](../../../samples/snippets/fsharp/azure/queue-storage.fsx#L60-L61)]
 
 ## Use Async workflows with common Queue storage APIs
 
 This example shows how to use an async workflow with common Queue storage APIs.
```
Comment (tg-msft) [inline docs/fsharp/using-fsharp-on-azure/queue-storage.md:123]: We should add a comment that all of the Queue methods we've shown so far have `Async` alternatives if customers would prefer the more performant option that doesn't block.

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md [hunk 18]
```diff
@@ -124,40 +116,40 @@ software failure, another instance of your code can get the same message
...
+time, so after 5 minutes have passed since the call to `ReceiveMessages`, any
 messages that have not been deleted will become visible again.
 
-[!code-fsharp[QueueStorage](~/samples/snippets/fsharp/azure/queue-storage.fsx#L97-L99)]
+[!code-fsharp[QueueStorage](../../../samples/snippets/fsharp/azure/queue-storage.fsx#L82-L84)]
 
 ## Get the queue length
 
-You can get an estimate of the number of messages in a queue. The `FetchAttributes` method asks the Queue service to retrieve the queue attributes, including the message count. The `ApproximateMessageCount` property returns the last value retrieved by the `FetchAttributes` method, without calling the Queue service.
+You can get an estimate of the number of messages in a queue. The `GetProperties` method asks the Queue service to retrieve the queue attributes, including the message count. The `ApproximateMessagesCount` property returns the last value retrieved by the `GetProperties` method, without calling the Queue service.
```
Comment (tg-msft) [inline docs/fsharp/using-fsharp-on-azure/queue-storage.md:143]: That "without calling the service" bit is no longer relevant in the latest client libraries.  (We still don't call the service, but NONE of our models will ever call the service.)

```suggestion
You can get an estimate of the number of messages in a queue. The `GetProperties` method asks the Queue service to retrieve the queue attributes, including the message count. The `ApproximateMessagesCount` property returns the last value retrieved by the `GetProperties` method.
```

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md [hunk 19]
```diff
@@ -124,40 +116,40 @@ software failure, another instance of your code can get the same message
...
 
 ## Delete a queue
 
 To delete a queue and all the messages contained in it, call the
 `Delete` method on the queue object.
 
-[!code-fsharp[QueueStorage](~/samples/snippets/fsharp/azure/queue-storage.fsx#L112-L113)]
+[!code-fsharp[QueueStorage](../../../samples/snippets/fsharp/azure/queue-storage.fsx#L97-L97)]
 
 ## Next steps
```
Comment (tg-msft) [inline docs/fsharp/using-fsharp-on-azure/queue-storage.md:154]: We should add a blurb that if you're migrating from the old libraries, they Base64 encoded messages by default but the new libraries do not because it's more performant.  See https://docs.microsoft.com/en-us/dotnet/api/azure.storage.queues.queueclientoptions.messageencoding?view=azure-dotnet#Azure_Storage_Queues_QueueClientOptions_MessageEncoding for how to set that up.

This is the most common problem people have when migrating to the new Queues library.

**Comment by zedy-wj** [comment]
Hi, @tg-msft Thanks for your review! Updated according to your comments.

**Review by BillWagner — APPROVED** [review]
All comments have been addressed.  

This LGTM, and I'll :shipit: now.

File: docs/fsharp/using-fsharp-on-azure/queue-storage.md [hunk 20]
```diff
@@ -109,60 +101,65 @@ you would keep a retry count as well, and if the message is retried more
...
 
 To delete a queue and all the messages contained in it, call the
 `Delete` method on the queue object.
 
-[!code-fsharp[QueueStorage](~/samples/snippets/fsharp/azure/queue-storage.fsx#L112-L113)]
+[!code-fsharp[QueueStorage](../../../samples/snippets/fsharp/azure/queue-storage.fsx#L99-L99)]
+
+## Note
+
+If you're migrating from the old libraries, they Base64 encoded messages by default but the new libraries do not because it's more performant.
```
Comment (BillWagner) [inline docs/fsharp/using-fsharp-on-azure/queue-storage.md:157]:
```suggestion
If you're migrating from the old libraries, they Base64 encoded messages by default but the new libraries do not because it's more performant.
```
